openwisp_controller/config/base/config.py

```diff
@@ -223,14 +223,13 @@ def manage_vpn_clients(cls, action, instance, pk_set, **kwargs):
         # coming from admin ModelForm
         else:
             templates = pk_set
-        # when clearing all templates
-        if action == 'post_clear':
-            for client in instance.vpnclient_set.all():
-                client.delete()
-            return
         # when adding or removing specific templates
         for template in templates.filter(type='vpn'):
             if action == 'post_add':
+                if vpn_client_model.objects.filter(
+                    config=instance, vpn=template.vpn
+                ).exists():
+                    return
```

Comment:
Shouldn't this be `continue` instead of `return`?

Reply:
mm I think you're right, but we have to verify, so how would this bug be triggered?

Maybe:
- assign 1 vpn template, save
- assign another vpn template, save

Maybe the second assignment won't work?

Can you please try?